infer.py

```diff
@@ -176,6 +196,19 @@ def _visualize_attention(self, model, beam_size, output_history, sliced_data, re
             '''
         print(interest_cnt * 1.0 / cnt)
 
+def chunked(items, size):
+    elements = []
+    for it in items:
+        elements.append(it)
+        if len(elements) == size:
+            yield elements
+            elements = []
+    if elements:
+        yield elements
+
+def listener(total, pbar_queue):
+    for _ in tqdm.tqdm(iter(pbar_queue.get, None), total=total, smoothing=0):
```

Comment:
```suggestion
    for _ in tqdm.tqdm(iter(pbar_queue.get, None), total=total, smoothing=0, dynamic_ncols=True):
```
`dynamic_ncols=True` resizes the progress bar if the window is resized (e.g. after reconnecting to the server), without it the bar won't draw correctly afterwards.

Reply:
fixed